deploy/examples/create-external-cluster-resources.py

```diff
@@ -554,9 +554,14 @@ def validate_rgw_endpoint_tls_cert(self):
                 return contents.rstrip()
 
     def _check_conflicting_options(self):
-        if not self._arg_parser.upgrade and not self._arg_parser.rbd_data_pool_name:
+        if (
+            not self._arg_parser.upgrade
+            and not self._arg_parser.rbd_data_pool_name
+            and not self._arg_parser.cephfs_filesystem_name
+            and not self._arg_parser.cephfs_data_pool_name
+        ):
             raise ExecutionFailureException(
-                "Either '--upgrade' or '--rbd-data-pool-name <pool_name>' should be specified"
+                "Either '--upgrade' or '--rbd-data-pool-name <pool_name>' or '--cephfs-filesystem-name <name>' or '--cephfs-data-pool-name <name>'  should be specified"
```

Comment:
we cannot have either `--cephfs-filesystem-name` or `--cephfs-data-pool-name` it should be both right?

Reply:
It is always not necessary https://github.com/rook/rook/blob/master/deploy/examples/create-external-cluster-resources.py#L1176

If not provided script can fetch the default values

Reply:
Is this a fair way to describe the behavior?
- If `--upgrade` is specified, it doesn't matter if the rbd or cephfs flags are specified
- Else
   - At least one of the rbd or cephfs flags must be specified, otherwise return an error

Question: What if they only want an external object store with no rbd or cephfs?

Reply:
https://github.com/rook/rook/pull/15028 we do have a pr in progress to enable it Even without Ceph cluster,
But we can include the --rgw-endpoint flag in the list too with this change...

Reply:
The function is currently named `_check_conflicting_options()`. This does not seem accurate. This function is instead checking for at least one required flag. 

What would be the behavior if we just completely removed this check? Then all flags would be optional, which seems like that's the request of the upstream issue.

Reply:
We don't want `upgrade` and creation at the same time, what if the user applied (--upgrade and creation) at the same time

Reply:
Upgrade and creation can be conflicting, thats why maybe the function name is like that

Reply:
For restricted user, the upgrade requires the data pool and filesystem flags too
```
kubectl -n rook-ceph exec rook-ceph-tools-6c6b9f7f-nqms9 -- python3 /etc/ceph/create-external-cluster-resources.py --upgrade --rbd-data-pool-name replicapool --k8s-cluster-name rookstorage --run-as-user client.csi-rbd-node-rookstorage-replicapool
```

The best way for now is to remove it,

But the filesystem needs to be re-looked, and I see a gap in how we upgrade restricted users, which can be improved.

Reply:
> Upgrade and creation can be conflicting, thats why maybe the function name is like that

IIUC, it was valid to have both the upgrade and rbd pool flags specified. If it was a conflict, I would expect them not to be allowed at the same time. But anyway, looks good to just remove this check and follow up if more specific changes are needed.